initramfs/init.sh

```diff
@@ -180,42 +180,81 @@ isBootingRestoreImage()
     findfs LABEL=recovery_data 
 }
 
-check_and_set_article_number()
+check_and_set_eeprom_data()
 {
+    # The printer's mainboard EEPROM can be "provisioned" by inserting a USB drive during boot.
+    # The USB drive must contain a single ext4 partition, containing one or more of the following
+    # files:
+    #
+    # - article_number
+    #   A single-line text file, containing the printer's BOM number in hex notation, exactly 4 bytes.
+    #   For instance, for an S5R2 printer, the file should contain the following line:
+    #       0x00 0x03 0x45 0xCB
+    #   (this is equivalent to decimal 214475).
+    #
+    # - country_code_lock
+    #   A single line text file, containing the printer's locked country code, in hex notation,
+    #   exactly 2 bytes.
+    #   For instance, to lock the printer to the AU country code, the file should contain the line:
+    #     0x41 0x55
+    #   To clear the lock, the file should contain:
+    #     0xFF 0xFF
+    #
+    # If either of these files is missing, no data is written to that part of the EEPROM.
+    #
+    # Relevant EEPROM layout here:
+    # (See opinicus for full details on EEPROM layout)
+    #   0x0100 - 0x0104     BOM number / article number
+    #   ...
+    #   0x0118 - 0x0120     Locked country code
+
     dev="/dev/sda1"
-    article_number_file="${ARTICLENR_USB_MOUNT}/article_number"
+    article_number_file="${PROVISIONING_USB_MOUNT}/article_number"
+    country_code_lock_file="${PROVISIONING_USB_MOUNT}/country_code_lock"
 
-    retries=5
-    while [ "${retries}" -gt 0 ]; do
-        if [ ! -b "${dev}" ]; then
-            retries="$((retries - 1))"
-            sleep 1
-            continue
-        fi
+    # Get the article number from EEPROM
+    art_num=$(i2ctransfer -y 3 w2@0x57 0x01 0x00 r4)
+    echo "---> Article number read from EEPROM: >${art_num}<"
 
-        echo "Attempting to mount '${dev}'."
-        if ! mount -t f2fs,ext4,vfat,auto -o exec,noatime "${dev}" "${ARTICLENR_USB_MOUNT}"; then
-            return 0
-        fi
+    # Get the country code lock from EEPROM
+    country_code_lock=$(i2ctransfer -y 3 w2@0x57 0x01 0x18 r2)
+    echo "--> Country code lock read from EEPROM: >${country_code_lock}<"
 
-        if [ ! -r "${article_number_file}" ]; then
-            umount "${dev}"
-            echo "No article number file found on '${dev}', skipping."
-            return 0
-        fi
+    # Wait for the USB drive to become visible; max 5 seconds.
+    retries=5
+    while [ ! -b "${dev}" ] && [ "${retries}" -gt 0 ]; do
+        retries="$((retries - 1))"
+        sleep 1
+    done
 
+    echo "Attempting to mount '${dev}'."
+    if ! mount -t f2fs,ext4,vfat,auto -o exec,noatime "${dev}" "${PROVISIONING_USB_MOUNT}"; then
```

Comment:
I know the original code didnt have it, but I think it is better to mount the USB Stick as Read Only, specially because it can be FAT.
```suggestion
    if ! mount -t f2fs,ext4,vfat,auto -o ro,exec,noatime "${dev}" "${PROVISIONING_USB_MOUNT}"; then
```